src/uu/sort/src/sort.rs

```diff
@@ -252,7 +250,7 @@ impl NumCache {
 #[derive(Clone)]
 struct Selection {
     range: SelectionRange,
-    num_cache: NumCache,
+    num_cache: Option<Box<NumCache>>,
```

Comment:
`Option<Box<NumCache>>` only takes up one word (i.e. 8 bytes) because the `None` case is represented by a null pointer. This adds a layer of indirection when we actually use the `NumCache` for numeric comparisons, but I think that's a good tradeoff, especially since we're still faster than GNU sort for numeric comparisons (but use twice as much memory).